### DISCUSSION TITLE
Base64 Encoded JSON Endpoint Rules are quite wasteful

### DISCUSSION THREAD

**Discussion by casperisfine** [aws/aws-sdk-ruby#2816] [Q&A]
We profile our application boot nightly and I discovered what I'd consider a regression after upgrading `aws-sdk`.

Firs the Base64 encoding seem like an useless step for what is generated code.

Similarly the JSON parsing, while easier to understand, seem a bit superfluous and cause a lot of duplicated strings for little reasons where a Ruby hash literal would have avoided this.

I managed to track the change to https://github.com/aws/aws-sdk-ruby/pull/2741, but I don't understand the rationale.

I'm curious what's the reason for this chance, and whether you'd be open to a PR that would change this for some decoded JSON, or best some Ruby literals?

**Comment by mullermp** [comment]
Thanks for opening up a discussion. Yes, the base64 encoded rules are not ideal. It boils down to rushed deadlines. @alextwoods has a draft to use literals instead.

**Reply by casperisfine** [reply]
Ah thank you very much, I'll subscribe to https://github.com/aws/aws-sdk-ruby/pull/2815. Let me know if I can help.
